core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -269,6 +269,26 @@ private[spark] object Utils extends Logging {
     file.setExecutable(true, true)
   }
 
+  /**
+   * Move data to trash on truncate table given
+   * 'spark.sql.truncate.trash.enabled' is true
+   */
+  def moveToTrashIfEnabled(
+      fs: FileSystem,
+      partitionPath: Path,
+      isTrashEnabled: Boolean,
+      hadoopConf: Configuration): Unit = {
```

Comment:
Let's have a return value here instead of `Unit` for the future.
1. If `isTrashEnabled=false`, this function had better return the return value of `fs.delete(partitionPath, true)`.
2. If `isTrashEnabled=true`, 
   - If `isSuccess=true`, `true` is return.
   - If `isSuccess=false`, the result of `fs.delete(partitionPath, true)` is return again.